src/parser/esprima_cpp/esprima.cpp

```diff
@@ -5723,86 +5723,64 @@ class Parser : public gc {
-        /*
         MetaNode node = this->createNode();
-        const isGenerator = true;
-        const previousAllowYield = this->context.allowYield;
+        const bool previousAllowYield = this->context->allowYield;
+        this->expect(LeftParenthesis);
 
-        this->context.allowYield = true;
-        const params = this->parseFormalParameters();
-        this->context.allowYield = false;
-        const method = this->parsePropertyMethod(params);
-        this->context.allowYield = previousAllowYield;
+        pushScopeContext(AtomicString());
+        this->context->allowYield = true;
+        ParseFormalParametersResult formalParameters = this->parseFormalParameters();
+        this->context->allowYield = false;
+        RefPtr<Node> method = this->isolateCoverGrammar(&Parser::parseFunctionSourceElements);
+        this->context->allowYield = previousAllowYield;
 
-        return this->finalize(node, new Node.FunctionExpression(null, params.params, method, isGenerator));
-        */
+        extractNamesFromFunctionParams(formalParameters.params);
+        scopeContexts.back()->m_paramsStart.index = node.index;
+        return this->finalize(node, new FunctionExpressionNode(AtomicString(), std::move(formalParameters.params), method.get(), popScopeContext(node), true));
     }
 
     // ECMA-262 14.4 Generator Function Definitions
 
-    Node* parseYieldExpression()
+    template <typename T, bool isParse>
+    T yieldExpression()
     {
-        this->throwError("yield keyword is not supported yet");
-        RELEASE_ASSERT_NOT_REACHED();
-        /*
-        const node = this->createNode();
-        this->expectKeyword('yield');
+        MetaNode node = this->createNode();
+        this->expectKeyword(YieldKeyword);
+
+        RefPtr<Node> exprNode;
+        bool delegate = false;
 
-        let argument = null;
-        let delegate = false;
         if (!this->hasLineTerminator) {
-            const previousAllowYield = this->context.allowYield;
-            this->context.allowYield = false;
-            delegate = this->match('*');
-            if (delegate) {
+            const bool previousAllowYield = this->context->allowYield;
+            this->context->allowYield = false;
+            delegate = this->match(Multiply);
+
+            if (delegate == true) {
                 this->nextToken();
-                argument = this->assignmentExpression<Parse>();
+                if (isParse == true) {
+                    exprNode = this->assignmentExpression<Parse>();
+                } else {
+                    this->assignmentExpression<Scan>();
+                }
             } else {
-                if (!this->match(';') && !this->match('}') && !this->match(')') && this->lookahead.type !== Token.EOF) {
-                    argument = this->assignmentExpression<Parse>();
+                if (!this->match(SemiColon) && !this->match(RightBrace) && !this->match(RightParenthesis) && this->lookahead.type != Token::EOFToken) {
```

Comment:
For readability / State expression explicitly
if (this->match(SemiColon) == false && this->match(RightBrace) == false && this->match(RightParenthesis) == false && this->lookahead.type != Token::EOFToken)